src/main/scala/io/iohk/ethereum/db/dataSource/RocksDbDataSource.scala

```diff
@@ -0,0 +1,248 @@
+        withResources(new WriteBatch()){ batch =>
+          toRemove.foreach{ key => batch.delete(key) }
+          toUpsert.foreach{ case (k, v) => batch.put(k, v) }
+
+          db.write(writeOptions, batch)
+        }
+      }
+    } catch {
+      case e: Exception =>
+        logger.error(s"DataSource not updated (toRemove: ${ toRemove.size }, toUpsert: ${ toUpsert.size }), cause: {}", e.getMessage)
+    } finally {
+      RocksDbDataSource.dbLock.readLock().unlock()
+    }
+    this
+  }
+
+  /**
+    * This function updates the DataSource by deleting all the (key-value) pairs in it.
+    *
+    * @return the new DataSource after all the data was removed.
+    */
+  override def clear: DataSource = {
+    destroy()
+    logger.debug(s"About to create new DataSource for path: ${ rocksDbConfig.path }")
+    val (newDb, readOptions) = RocksDbDataSource.createDB(rocksDbConfig)
+    this.db = newDb
+    this.readOptions = readOptions
+    this
+  }
+
+  /**
+    * This function closes the DataSource, without deleting the files used by it.
+    */
+  override def close(): Unit = {
+    logger.debug(s"About to close DataSource in path: ${ rocksDbConfig.path }")
+    RocksDbDataSource.dbLock.writeLock().lock()
+    try {
+      db.close()
+    } catch {
+      case e: Exception =>
+        logger.error("Not closed the DataSource properly, cause: {}", e)
+    } finally {
+      RocksDbDataSource.dbLock.writeLock().unlock()
+    }
+    readOptions.close()
+  }
+
+  /**
+    * This function closes the DataSource, if it is not yet closed, and deletes all the files used by it.
+    */
+  override def destroy(): Unit = {
+    try {
+      close()
+    } finally {
+      import rocksDbConfig._
+
+      val options = new Options()
+        .setCreateIfMissing(createIfMissing)
+        .setParanoidChecks(paranoidChecks)
+        .setCompressionType(CompressionType.LZ4_COMPRESSION)
+        .setBottommostCompressionType(CompressionType.ZSTD_COMPRESSION)
+        .setLevelCompactionDynamicLevelBytes(true)
+        .setMaxOpenFiles(maxOpenFiles)
+        .setIncreaseParallelism(maxThreads)
+
+      logger.debug(s"About to destroy DataSource in path: $path")
+      RocksDB.destroyDB(path, options)
+    }
+  }
+}
+
+trait RocksDbConfig {
+  val createIfMissing: Boolean
+  val paranoidChecks: Boolean
+  val path: String
+  val maxThreads: Int
+  val maxOpenFiles: Int
+  val verifyChecksums: Boolean
+  val levelCompaction: Boolean
+}
+
+object RocksDbDataSource {
+
+  /**
+    * This lock is needed because close and open operations in RocksDb are not thread-safe.
+    */
+  private val dbLock = new ReentrantReadWriteLock()
+
+  private def createDB(rocksDbConfig: RocksDbConfig): (RocksDB, ReadOptions) = {
+    import rocksDbConfig._
+
+    RocksDB.loadLibrary()
+
+    RocksDbDataSource.dbLock.writeLock().lock()
+
+    val readOptions = new ReadOptions().setVerifyChecksums(rocksDbConfig.verifyChecksums)
+    try {
+      val options = new Options()
```

Comment:
I think we can add here this other table options https://github.com/ethereum/ethereumj/blob/develop/ethereumj-core/src/main/java/org/ethereum/datasource/rocksdb/RocksDbDataSource.java#L122, as they are really improving performnace of rocks db.

Reply:
what do you think about putting all that into config of rocksdb?

Reply:
i think that maybe blockcache size should be configurable, as it will be beneficial to user who has bigger memory supply. But as for other options i think they can be hardocoded, as it is hard to imagine that user would want to to go docs and read what bloom filter does or any of the other options.